src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<SyntaxNode>
+        {
+            private List<ExpressionSyntax> _assignmentTargetsOpt;
+
+            private Rewriter()
+            {
+            }
+
+            public static ExpressionSyntax Rewrite(SwitchStatementSyntax node, out List<ExpressionSyntax> assignmentTargetsOpt)
+            {
+                var rewriter = new Rewriter();
+                var result = rewriter.VisitSwitchStatement(node);
+                assignmentTargetsOpt = rewriter._assignmentTargetsOpt;
+                return (ExpressionSyntax)result;
```

Comment:
seems like an out-param isn't really necessary here.

Reply:
rewrite operates on node, targets are addition info, so I'd prefer out param instead of tuple.for it.